implicits/src/upickle/implicits/internal/Macros.scala

```diff
@@ -209,8 +209,16 @@ object Macros {
     def annotate(tpe: c.Type)(derived: c.universe.Tree) = {
       val sealedParent = tpe.baseClasses.find(_.asClass.isSealed)
       sealedParent.fold(derived) { parent =>
-        val index = customKey(tpe.typeSymbol).getOrElse(tpe.typeSymbol.fullName)
-        q"${c.prefix}.annotate($derived, $index)"
+        val tagName = customDiscriminator(parent) match {
+          case Some(customName) => Literal(Constant(customName))
+          case None =>
+            c.prefix.actualType.members.find(_.name == TermName("tagName")) match {
```

Comment:
Additional complexity here vs https://github.com/rallyhealth/weePickle/pull/13 to handle the legacyApi. Tests pass, but no idea how robust this approach is. Please double check.